WatchList Modern Charts - rendering selected points fix

radek2s opened the pull request:
Modern WatchList now will be able to render only selected points from the WatchList. 
It has been fixed and point change also change the chart series that are rendered.

Limraj reviewed (commented):
Summarizing the tests:
1) Modern watchlist, by default all data series from the watchlist are selected and when loading the view it generates graphs for all series, on the old watchlist it generates graphs only after clicking on the button;
2) Charts are updated only after deselecting selected series and clicking refresh, in the old watchlist, the charts are updated automatically after deselecting a given series;
3) Fixed that you can limit the number of data series in the chart via the refresh button;
4) Added use API to aggregate multiple values ​​for a point;
5) Incremental loading of the chart, and aggregation api, it means that some of the data on the chart is aggregated and new values ​​are not aggregated.
6) The y axis is not scaled like in the old chart, i.e. if we have a series of data with random values ​​within 0-1, and the second series within 100-200, the graph of the first data series reduces to a straight line, both data series are drawn to the same scale.

radek2s commented:
> 1. Modern watchlist, by default all data series from the watchlist are selected and when loading the view it generates graphs for all series, on the old watchlist it generates graphs only after clicking on the button;

Yes, but the request from the customer was to keep the chart configuration after changing the active WatchList. So if you once create a chart for specific watch list it should be always rendered when it is visible with the last configuration.

> 2. Charts are updated only after deselecting selected series and clicking refresh, in the old watchlist, the charts are updated automatically after deselecting a given series;

Yes, that is true. That has been changed in Vue.js version of that page (it is done in [this PR](https://github.com/SCADA-LTS/Scada-LTS/pull/2009)) because to integrate the pure JavaScript method invocation with Vue.js application is more complex task, and resource consuming. In the PR#2009 it was done by StateManager in Vue.js that watch the selected points and on their change the graph is updated too. But here I've prepared the fastest and easiest solution. But if there is a need to use it in other way I can implement that feature. 

> 3. Fixed that you can limit the number of data series in the chart via the refresh button;

That is how it was designed to be working. When user make any change on the watchlist the chart won't be updated unless he click the "refresh button". That is caused due to performance issues because that chart is loading much slower than this "old one" that was just a *.png graphic from the server. Each reload require to fetch the new set of DataPoint Values from the API. So to limit that operations it is done only after clicking refresh button.

> 4. Added use API to aggregate multiple values ​​for a point;
> 5. Incremental loading of the chart, and aggregation api, it means that some of the data on the chart is aggregated and new values ​​are not aggregated. 

API should be integrated for all AmChart instances but yes you are right I have not enabled that feature in this WatchList chart version. I will enable it in next commits. On the left side is the method that enable that feature, and on the right is client code that use this class to create a chart and that method is missing. 
![image](https://user-images.githubusercontent.com/22638815/147912366-8f90afe6-b1f6-40c2-972c-38eeab56ad2b.png)

> 6. The y axis is not scaled like in the old chart, i.e. if we have a series of data with random values ​​within 0-1, and the second series within 100-200, the graph of the first data series reduces to a straight line, both data series are drawn to the same scale.

Yes, but that also was done in the first version of the ModernCharts where it was described that each series is by default assigned to the first Y axis. But there are more than one Y axis and by changing the chart settings you are able to move specific data point series and assign it into another Y axis. Just open the settings button and change the Y axis from "1" to "2".
